### PR TITLE
366015: Update Spring + Spring Boot

### PR DESCRIPTION
**Pull request by rorytorneymf**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=366015

See: https://github.com/spring-projects/spring-boot/issues/30971

### CODE REVIEW

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingApi.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    @ApiOperation(value = "Upload documents. The batch will be automatically created if it doesn't already exist.", nickname = "createOrReplaceBatch", notes = "", tags={  })
+    @ApiResponses(value = { 
+        @ApiResponse(code = 200, message = "Successfully uploaded batch of documents."),
+        @ApiResponse(code = 400, message = "The request could not be processed because one or more arguments are invalid."),
+        @ApiResponse(code = 500, message = "The request failed due to an unexpected server error.") })
+    @RequestMapping(value = "/batches/{batchId}",
+        consumes = { "multipart/mixed" },
+        method = RequestMethod.PUT)
+    ResponseEntity<Void> createOrReplaceBatch(@ApiParam(value = "Identifies the tenant making the request." ,required=true) @RequestHeader(value="X-TENANT-ID", required=true) String X_TENANT_ID,@Size(min=1) @ApiParam(value = "Identifies the batch.",required=true) @PathVariable("batchId") String batchId,@ApiParam(value = ""  ) MultipartHttpServletRequest request);
```
Comment (rorytorneymf) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingApi.java]: Copied and modified in order to change `@Valid @RequestBody Object body` to `MultipartHttpServletRequest request`. See https://github.com/spring-projects/spring-boot/issues/30971

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java [hunk 2]
```diff
@@ -147,35 +146,26 @@ public List<String> saveFiles(final TenantId tenantId, BatchId batchId, FileItem
...
-                    fileItemStream = fileItemIterator.next();
-                } catch (FileUploadException | IOException ex) {
-                    throw new IncompleteBatchException(ex);
-                }
+            LOGGER.debug("Retrieving next part...");
+
+            for (final Part part : parts) {
 
-                if (!fileItemStream.isFormField()) {
+                if (part.getSubmittedFileName() != null) {
```
Comment (rorytorneymf) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/dao/filesystem/FileSystemDao.java:154]: I think `part.getSubmittedFileName() != null` is equivalent to `!fileItemStream.isFormField()`

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingApi.java (hunk 1 above)
Comment (rorytorneymf) [reply staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingApi.java]: Actually, I remember there's a way to do this in maven using the `gmavenplus-plugin`, I'll have a look at this on Monday

**Review by dermot-hardy — CHANGES_REQUESTED** [review]
I tried testing this and I'm finding that it abruptly disconnects if I send it large files.  I see the following in the log:

    Request processing failed; nested exception is org.springframework.web.multipart.MaxUploadSizeExceededException: Maximum upload size exceeded; nested exception is java.lang.IllegalStateException: org.apache.tomcat.util.http.fileupload.impl.FileSizeLimitExceededException: The field batch.json exceeds its maximum permitted size of 1048576 bytes.

I'm concerned that this is more than just a maximum size setting that needs to be adjusted.  The Staging Service is meant to stream the data through to disk, so that it never has to hold the entire request in memory at any one time.  This feels like it is trying to parse the entire request before it even calls into our code.

**Comment by rorytorneymf** [comment]
> I tried testing this and I'm finding that it abruptly disconnects if I send it large files. I see the following in the log:
> 
> ```
> Request processing failed; nested exception is org.springframework.web.multipart.MaxUploadSizeExceededException: Maximum upload size exceeded; nested exception is java.lang.IllegalStateException: org.apache.tomcat.util.http.fileupload.impl.FileSizeLimitExceededException: The field batch.json exceeds its maximum permitted size of 1048576 bytes.
> ```
> 
> I'm concerned that this is more than just a maximum size setting that needs to be adjusted. The Staging Service is meant to stream the data through to disk, so that it never has to hold the entire request in memory at any one time. This feels like it is trying to parse the entire request before it even calls into our code.

Thanks. I've gone back to using `commons-fileupload` in this new PR:

https://github.com/CAFDataProcessing/staging-service/pull/115

Closing this PR.

**Pull request closed without merging**